lib/pry/method/weird_method_locator.rb

```diff
@@ -77,15 +81,31 @@ def target_self
       end
 
       def target_file
-        pry_file? ? target.eval('__FILE__') : File.expand_path(target.eval('__FILE__'))
+        file =
+          if target.respond_to?(:source_location)
+            target.source_location.first
+          else
+            target.eval('__FILE__')
+          end
+        pry_file? ? file : File.expand_path(file)
       end
 
       def target_line
-        target.eval('__LINE__')
+        if target.respond_to?(:source_location)
+          target.source_location.last
+        else
+          target.eval('__LINE__')
+        end
       end
 
       def pry_file?
-        Pry.eval_path == target.eval('__FILE__')
+        file =
+          if target.respond_to?(:source_location)
```

Comment:
This condition is similar to the line `84`. I think you could move this in a function

```rb
def target_line
  source_location(target)
end

def pry_file?
  file = source_location(target)
  Pry.eval_path == file
end

def source_location(target)
  if target.respond_to?(:source_location)
    target.source_location.first
  else
    target.eval('__FILE__')
  end
end
```

WDYT? 🤔

Reply:
If I were writing this code myself, I would strongly consider building it that way, but I was hoping to change as little as possible since I am just backporting someone else's fix.